source/cgal/surface_mesh.cc

```diff
@@ -114,10 +127,60 @@ namespace CGALWrappers
                     (f % 2 == 0));
   }
 
-  // explicit instantiations
-#    include "surface_mesh.inst"
 
 
+  template <typename CGALPointType, int dim, int spacedim>
+  void
+  dealii_tria_to_cgal_surface_mesh(
+    const dealii::Triangulation<dim, spacedim> &tria,
+    const Mapping<dim, spacedim> &              mapping,
+    CGAL::Surface_mesh<CGALPointType> &         mesh)
+  {
+    Assert(tria.n_cells() > 0,
+           ExcMessage(
+             "Triangulation cannot be empty upon calling this function."));
+    Assert(mesh.is_empty(),
+           ExcMessage(
+             "The surface mesh must be empty upon calling this function."));
+    Assert((dynamic_cast<const MappingQ<dim, spacedim> *>(&mapping) != nullptr),
+           ExcMessage("Not yet implemented for this Mapping."));
```

Comment:
You are not using the mapping. Remove it from the input arguments, and remove this assert.